blockchain/state/sync_test.go

```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
 	return it.Error
 }
 
+// compareStatesConsistency checks that all data of given two states.
+func compareStatesConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash) error {
+	// Create and iterate a state trie rooted in a sub-node
+	if _, err := oldDB.ReadStateTrieNode(root.Bytes()); err != nil {
+		return err
+	}
+	oldState, err := New(root, NewDatabase(oldDB))
+	if err != nil {
+		return err
+	}
+
+	if _, err := newDB.ReadStateTrieNode(root.Bytes()); err != nil {
+		return err
+	}
+	newState, err := New(root, NewDatabase(newDB))
+	if err != nil {
+		return err
+	}
+
+	oldIt := NewNodeIterator(oldState)
+	newIt := NewNodeIterator(newState)
+	for oldIt.Next() {
+		if !newIt.Next() {
+			return fmt.Errorf("newDB iterator finished early : oldIt.Hash(%v) oldIt.Parent(%v)", oldIt.Hash, oldIt.Parent)
+		}
+
+		if oldIt.Hash != newIt.Hash {
+			return fmt.Errorf("mismatched hash oldIt.Hash : oldIt.Hash(%v) newIt.Hash(%v)", oldIt.Hash, newIt.Hash)
+		}
+
+		if oldIt.Parent != newIt.Parent {
+			return fmt.Errorf("mismatched parent hash : oldIt.Parent(%v) newIt.Parent(%v)", oldIt.Parent, newIt.Parent)
+		}
```

Comment:
Why don't you check other field values such as `code`?

Reply:
Good point. I will add. :)